File: Astronomy-objc/Astronomy-objc/MarsPhotoReference.m
//
//  MarsPhotoReference.m
//  Astronomy-objc
//
//  Created by Matthew Martindale on 8/5/20.
//  Copyright © 2020 Matthew Martindale. All rights reserved.
//

#import "MarsPhotoReference.h"

@implementation MarsPhotoReference

- (instancetype)initWithId:(int)photoId
            imageURLString:(NSString *)imageURLString
{
    if (self = [super init]) {
        _photoId = photoId;
        _imageURLString = imageURLString;
    }
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dictionary
{
    NSNumber *photoId = [dictionary objectForKey:@"id"];
    if (![photoId isKindOfClass:NSNumber.class]) return nil;
    
    NSString *imageURLString = [dictionary objectForKey:@"img_src"];
    if (![imageURLString isKindOfClass:NSString.class]) return nil;
    
    return [self initWithId:photoId.intValue imageURLString:imageURLString];
}

@end

File: Astronomy-ObjC/Astronomy-ObjC/MarsPhotoReference.swift
//
//  MarsPhotoReference.swift
//  Astronomy-ObjC
//
//  Created by John Kouris on 2/12/20.
//  Copyright © 2020 John Kouris. All rights reserved.
//

import Foundation

@objc class MarsPhotoReference: NSObject {
    @objc let id: Int
    @objc let sol: Int
    @objc let imageURL: URL
    @objc let dateString: String
    
    @objc init(id: Int, sol: Int, imageURL: URL, dateString: String) {
        self.id = id
        self.sol = sol
        self.imageURL = imageURL
        self.dateString = dateString
    }
    
    @objc init?(with dictionary: [String: Any]) {
        guard let id = dictionary["id"] as? Int,
            let sol = dictionary["sol"] as? Int,
            let imageString = dictionary["img_src"] as? String,
            let imageURL = URL(string: imageString),
            let dateString = dictionary["earth_date"] as? String else {
            return nil
        }
        
        self.id = id
        self.sol = sol
        self.imageURL = imageURL
        self.dateString = dateString
        super.init()
    }
}
